core/chains/evm/gas/rollups/op_price_reader_test.go

```diff
@@ -0,0 +1,200 @@
+
+			l1BaseFeeMethodAbi, err := abi.JSON(strings.NewReader(L1BaseFeeAbiString))
+			require.NoError(t, err)
+			l1BaseFeeCalldata, err := l1BaseFeeMethodAbi.Pack(OPStackGasOracle_l1BaseFee)
+			require.NoError(t, err)
+
+			isEcotoneMethodAbi, err := abi.JSON(strings.NewReader(OPIsEcotoneAbiString))
+			require.NoError(t, err)
+			isEcotoneCalldata, err := isEcotoneMethodAbi.Pack(OPStackGasOracle_isEcotone)
+			require.NoError(t, err)
+
+			ethClient := mocks.NewETHClient(t)
+			call := ethClient.On("CallContract", mock.Anything, mock.IsType(ethereum.CallMsg{}), mock.IsType(&big.Int{})).Run(func(args mock.Arguments) {
+				callMsg := args.Get(1).(ethereum.CallMsg)
+				blockNumber := args.Get(2).(*big.Int)
+				require.Equal(t, isEcotoneCalldata, callMsg.Data)
+				require.Equal(t, oracleAddress, callMsg.To.String())
+				assert.Nil(t, blockNumber)
+			})
+
+			if tc.returnBadData {
+				call.Return([]byte{0x2, 0x2}, nil).Once()
+			} else if tc.isEcotoneError {
+				call.Return(nil, fmt.Errorf("test error")).Once()
+			} else {
+				call.Return(isEcotoneMethodAbi.Methods["isEcotone"].Outputs.Pack(false)).Once()
+			}
+
+			if !tc.returnBadData {
+				ethClient.On("CallContract", mock.Anything, mock.IsType(ethereum.CallMsg{}), mock.IsType(&big.Int{})).Run(func(args mock.Arguments) {
+					callMsg := args.Get(1).(ethereum.CallMsg)
+					blockNumber := args.Get(2).(*big.Int)
+					require.Equal(t, l1BaseFeeCalldata, callMsg.Data)
+					require.Equal(t, oracleAddress, callMsg.To.String())
+					assert.Nil(t, blockNumber)
+				}).Return(common.BigToHash(l1BaseFee).Bytes(), nil).Once()
+			}
+
+			oracle := newOPPriceReader(logger.Test(t), ethClient, config.ChainOptimismBedrock, oracleAddress)
+			gasPrice, err := oracle.GetDAGasPrice(testutils.Context(t))
+
+			if tc.returnBadData {
+				assert.Error(t, err)
+			} else {
+				require.NoError(t, err)
+				assert.Equal(t, l1BaseFee, gasPrice)
+			}
+		})
+	}
+}
+
+func setupIsEcotone(t *testing.T, oracleAddress string) *mocks.ETHClient {
+	isEcotoneMethodAbi, err := abi.JSON(strings.NewReader(OPIsEcotoneAbiString))
+	require.NoError(t, err)
+	isEcotoneCalldata, err := isEcotoneMethodAbi.Pack(OPStackGasOracle_isEcotone)
+	require.NoError(t, err)
+
+	ethClient := mocks.NewETHClient(t)
+	ethClient.On("CallContract", mock.Anything, mock.IsType(ethereum.CallMsg{}), mock.IsType(&big.Int{})).Run(func(args mock.Arguments) {
+		callMsg := args.Get(1).(ethereum.CallMsg)
+		blockNumber := args.Get(2).(*big.Int)
+		require.Equal(t, isEcotoneCalldata, callMsg.Data)
+		require.Equal(t, oracleAddress, callMsg.To.String())
+		assert.Nil(t, blockNumber)
+	}).Return(isEcotoneMethodAbi.Methods["isEcotone"].Outputs.Pack(true)).Once()
+
+	return ethClient
+}
+
+func TestDAPriceReader_ReadEcotoneGasPrice(t *testing.T) {
+	l1BaseFee := big.NewInt(100)
+	oracleAddress := common.HexToAddress("0x1234").String()
+
+	t.Parallel()
+
+	t.Run("Calling l1BaseFee when chain has not made Ecotone upgrade", func(t *testing.T) {
```

Comment:
I think this test's name is incorrect? Doesn't this method test when Ecotone is enabled?